### PR TITLE
Feature: test for User-agent header being too long

### PR DESCRIPTION
**Pull request by lwojtow**
* Add option to set custom 'User-Agent' header for a handshake
* Add test for large User-Agent header used in a handshake

### CODE REVIEW

File: src/tools/synth_node.rs [hunk 1]
```diff
@@ -33,7 +33,7 @@ pub fn enable_tracing() {
 }
 
 pub struct SyntheticNode {
-    inner: InnerNode,
+    pub inner: InnerNode,
```
Comment (danielvladco) [inline src/tools/synth_node.rs:36]: should be private

Comment (lwojtow) [reply src/tools/synth_node.rs]: Done

File: src/tests/resistance/handshake.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#[tokio::test]
+async fn r001_node_should_reject_and_survive_handshake_with_too_long_User_Agent_header() {
+    // ZG-RESISTANCE-001
+
+    // Build and start the Ripple node.
+    let target = TempDir::new().expect("Can't build tmp dir");
+    let mut node = Node::builder()
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .expect("Unable to start node");
```
Comment (Rqnsom) [inline src/tests/resistance/handshake.rs:18]:
```suggestion
        .expect("unable to start the node");
```

File: src/tests/resistance/handshake.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    tools::{config::TestConfig, synth_node::SyntheticNode},
+};
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_node_should_reject_and_survive_handshake_with_too_long_User_Agent_header() {
+    // ZG-RESISTANCE-001
+
+    // Build and start the Ripple node.
+    let target = TempDir::new().expect("Can't build tmp dir");
```
Comment (Rqnsom) [inline src/tests/resistance/handshake.rs:14]:
```suggestion
    let target = TempDir::new().expect("Couldn't create a temporary directory");
```

File: src/tests/resistance/handshake.rs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    let mut node = Node::builder()
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .expect("Unable to start node");
+
+    // Start the first synthetic node with a 'User-Agent' header that's too long.
+    let mut test_config = TestConfig::default();
+    test_config.synth_node_config.user_agent = format!("{:8192}", 0);
```
Comment (Rqnsom) [inline src/tests/resistance/handshake.rs:22]: > "{:8192}", 0

Aaa, TIL, nice!

File: src/tests/resistance/handshake.rs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    assert!(!synth_node1.is_connected(node.addr()));
+
+    // Start the second synthetic node with the default 'User-Agent'.
+    let synth_node2 = SyntheticNode::new(&Default::default()).await;
+    synth_node2.connect(node.addr()).await.unwrap();
+    // Ensure this connection was successful.
+    assert_eq!(synth_node2.num_connected(), 1);
+    assert!(synth_node2.is_connected(node.addr()));
+
+    // Shutdown both nodes.
```
Comment (Rqnsom) [inline src/tests/resistance/handshake.rs:36]: 😀
```suggestion
    // Shutdown all nodes.
```

**Review by Rqnsom — APPROVED** [review]
a few suggestions, otherwise LGTM!

File: src/tests/resistance/handshake.rs (hunk 2 above)
Comment (lwojtow) [reply src/tests/resistance/handshake.rs:18]: Done

File: src/tests/resistance/handshake.rs (hunk 3 above)
Comment (lwojtow) [reply src/tests/resistance/handshake.rs:14]: Done

File: src/tests/resistance/handshake.rs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+use tempfile::TempDir;
+
+use crate::{
+    setup::node::{Node, NodeType},
+    tools::{config::TestConfig, synth_node::SyntheticNode},
+};
+
+#[allow(non_snake_case)]
+#[tokio::test]
+async fn r001_node_should_reject_and_survive_handshake_with_too_long_User_Agent_header() {
```
[the remark this answers is not in the document]
Comment (lwojtow) [reply src/tests/resistance/handshake.rs:10]: Done

File: SPEC.md [hunk 7]
```diff
@@ -302,4 +302,14 @@ The test index makes use of symbolic language in describing connection and messa
...
+
+## Resistance
+
+### ZG-RESISTANCE-001
+
+    The node rejects handshake when the 'User-Agent' header is too long (8192 bytes in this test).
+    The node should be able to accept connections after such request.
+    This test attempts a handshake with long 'User-Agent' header and ensures that the connection
+    wasn't successful. Then, it attempts a normal connection and ensures that the connection was
+    successful.
```
[the remark this answers is not in the document]
Comment (lwojtow) [reply SPEC.md:315]: Done

File: src/tests/resistance/handshake.rs (hunk 5 above)
Comment (lwojtow) [reply src/tests/resistance/handshake.rs:36]: Good catch. Done.

**Pull request merged**